lib/util/stream.js

```diff
@@ -106,11 +106,50 @@ class PartialPipe {
   pipe(out) { return reduce(((x, y) => x.pipe(y)), this.streams[0], this.streams.slice(1)).pipe(out); }
 }
 
+// allWithStreams() is very similar to Promise.all() and returns the same
+// result. The key difference is that if `values` contains a stream or a promise
+// that resolves to a stream, then the fate of the stream will be tied to the
+// fate of each promise within `values`. If any promise is rejected, then the
+// stream will be immediately destroyed.
+const allWithStreams = (values) => {
```

Comment:
I don't love the name `allWithStreams()`, but I couldn't think of anything better. 🤔